src/main/io/serial.h

```diff
@@ -77,14 +77,20 @@ typedef enum {
     BAUD_COUNT
 } baudRate_e;
 
-extern const uint32_t baudRates[];
+extern const uint32_t baudRates[BAUD_COUNT];
+
+#if SERIAL_UART_FIRST_INDEX == 0
+#define SERIAL_UART_INDEX_BITFLAG  0x80
+#else
+#define SERIAL_UART_INDEX_BITFLAG  0x0
+#endif
 
 // serial port identifiers are now fixed, these values are used by MSP commands.
 typedef enum {
     SERIAL_PORT_ALL = -2,
     SERIAL_PORT_NONE = -1,
-    // prepare for transition to SERIAL_PORT_UART0
-    SERIAL_PORT_UART_FIRST = 0,
+    SERIAL_PORT_START_INDEX = 0,
```

Comment:
maybe it would be best to use SERIAL_PORT_UART_FIRST = 50. It would prevent collision will old assignment, msp will work without changing code (configurator will need to map both to correct uart) and as a bonus, 0 will be unused after this, making zero initialization more useful. 
and SITL code needs fixing anyway.
p

Reply:
Funny thing - I was just thinking the same thing! Given our differing thoughts on using 0 as a magic number...

Reply:
now just rename SERIAL_PORT_START_INDEX to SERIAL_PORT_LEGACY_UART1 and it will be perfect ;-)

SITL got broken by this, but I'll post separate PR to fix it tomorrow (it is not directly related to this pr)

Reply:
I'm actually thinking we move to 50 start index for both cases, and support backward compatibility / legacy by mapping the values on the FC, ie value + 51 for parsing.

Latest configurator required and for talking to prior versions of FC configurator can add / remove by 51 its index.

Reply:
Lots of question to raise with this implementation:

- having a new range (50) seems odd.
- 20 would work too - as we never exceeded 10 UARTs.
- Anyways would love to have this using 1-20 range.
- Not sure where UART0 became zero index and why this is needed.

Reply:
It is an identifier rather than an index. Zero based is not relevant here.

Reply:
@blckmn : my idea was to allow overlap between old and new version, for example until new release. but eventually both will be moved, ideally with uart1=51 in both cases. 

IMO configurator only needs to map id to name, but it does not use it id for anything else. so new configurator will be backward compatible, no need to translate between old and new ids. 
cli uses names, so it is fine 
only saved config will be problematic, but new PG version and dump/restore will fix it.

@haslinghuis : 


> * having a new range (50) seems odd.

20 is vcp, 30 softserial  40 lpuart. 50 is first free ..
my refactoring tried to abstract all this - only first index and count is important.

> * Anyways would love to have this using 1-20 range.

it will overlap old uart assignment. it can cause problems  it is safer to use nonoverlapping range. and itakes configurator simpler
and not using 0 as valid port is small bonus.

> * Not sure where UART0 became zero index and why this is needed.

new platforms with uarts named from zero. rp2350

> * Can we start from here. Making index zero based ???

the idea is to make it safe and avoid problems where possible. From programmer pov, it should be SERIAL_PORT_UART1, without any specific value. But SERIAL_PORT_UART1 + 1 = SERIAL_PORT_UART2


> * Sorry for ignorance - need to understand the need for UART0 in the first place.

see above. Also it improves cli

Reply:
Thanks. Understand new range would be easier to implement. Is it really not possible to refactor current range using zero based index ?

This will need documentation update and having two UART ranges (zero based 50-70 and one based 1-20)

Reply:
The documentation update will be use names not numbers. This will need to occur anyway. 

There is still the issue of indexing for DMA and other items to be resolved.